lib/middleware/autoreload.js

```diff
@@ -30,10 +29,13 @@ module.exports = function(options) {
 
     // enable AutoReload
     if (options.autoreload) {
-        var watch = new gaze.Gaze(watches);
-
-        // exposing watch to browser middleware
-        options.watch = watch;
+        if (!options.watch) {
+            var watch = new gaze.Gaze(watches);
+            // exposing watch to browser middleware
+            options.watch = watch;
+        } else {
+            var watch = options.watch;
+        }
```

Comment:
Even though this will still work, you should define `watch` outside of the if/else statement.